### PR TITLE
Cosine-based specular env map LOD selection

### PR DESCRIPTION
**Pull request by bhouston**
Built upon the simplified lighting PR: #7324

I've implemented the cosine env map lod queries in this PR based on the ideas here: #7411

### CODE REVIEW

**Comment by mrdoob** [comment]
@WestLangley thoughts?

**Comment by WestLangley** [comment]
After things settle down, please.

**Comment by bhouston** [comment]
I've updated this to work with the latest DEV branch and the other changes.

(Although I am very confused as to why a number of changes by @mrdoob in the dev branch appear in this PR -- is this github screwing up? I've never seen this before?  Or is there a revert in the history?)

**Comment by bhouston** [comment]
The weirdest thing is that if I do a straight compare between these two branches, it doesn't show these @mrdoob commits on dev branch -- it is proper, so I am fairly confused here: 

https://github.com/mrdoob/three.js/compare/dev...bhouston:cosine_envMap_lod

**Comment by mrdoob** [comment]
I think it's github screwing up. It's been doing it for a while. I only look at the "Files Changed" section.

**Comment by mrdoob** [comment]
Which, actually, shows that you're including the builds somehow.

**Comment by bhouston** [comment]
I submitted some extra stuff by accident as well.  I'll remove it now.

**Comment by mrdoob** [comment]
Yeah. Try rebasing.

**Comment by bhouston** [comment]
I removed two changes that probably are not necessary in this PR (although I didn't remove any builds, because there doesn't seem to be any) and it seems to be focused now: https://github.com/mrdoob/three.js/compare/dev...bhouston:cosine_envMap_lod

**Comment by mrdoob** [comment]
There is still a lot of noise in "Files changed". Hard to review until that it's clean...
https://github.com/mrdoob/three.js/pull/7414/files

**Comment by bhouston** [comment]
Rebase done.

**Comment by WestLangley** [comment]
This PR looks OK to me.

**Comment by mrdoob** [comment]
Thanks guys!

File: src/renderers/shaders/ShaderChunk/bsdfs.glsl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-float calcLightAttenuation( float lightDistance, float cutoffDistance, float decayExponent ) {
+float calcLightAttenuation( const in float lightDistance, const in float cutoffDistance, const in float decayExponent ) {
```
Comment (bhouston) [inline src/renderers/shaders/ShaderChunk/bsdfs.glsl]: no need to have "const in" here really, it just leads to more verbose code and I've never seen it make a performance difference.

**Pull request merged**